src/coreclr/src/vm/nativeimage.cpp

```diff
@@ -42,16 +56,18 @@ NativeImage::NativeImage(PEImageLayout *pImageLayout, LPCUTF8 imageFileName)
     }
     CONTRACTL_END;
 
+    m_pAssemblyLoadContext = pAssemblyLoadContext;
     m_pImageLayout = pImageLayout;
     m_fileName = imageFileName;
     m_eagerFixupsHaveRun = false;
 }
 
-void NativeImage::Initialize(READYTORUN_HEADER *pHeader, LoaderAllocator *pLoaderAllocator, AllocMemTracker *pamTracker)
+void NativeImage::Initialize(READYTORUN_HEADER *pHeader, LoaderAllocator *pLoaderAllocator)
```

Comment:
Why did you remove the memory tracking here? It opens up a possibility for a memory leak if there are racing loads of the native image in `NativeImage::Open`

Reply:
Hmm, you're obviously right, I have apparently messed this up in my third refactoring of the native image dictionary during the review, will fix.

Reply:
Fixed in 5th commit.